system/services/admin/LoginService.cfc

```diff
@@ -677,21 +671,15 @@ component displayName="Admin login service" {
 				two_step_auth_key_in_use = true
 			} );
 
-			var loginRecordDao = $getPresideObject( "security_user_two_factor_login_record" );
-			var updated = loginRecordDao.updateData( filter={
-				  security_user = userId
-				, ip_address    = arguments.ipAddress
-				, user_agent    = arguments.userAgent
-			}, data={ logged_in_date=Now() } );
-
-			if ( !updated ) {
-				loginRecordDao.insertData({
-					  security_user  = userId
-					, ip_address     = arguments.ipAddress
-					, user_agent     = arguments.userAgent
-					, logged_in_date = Now()
-				});
-			}
+			var cookieValue = CreateUUID();
```

Comment:
as per note above, this could be refactored:

```cfc
var loginRecordId = $getPresideObject( "security_user_two_factor_login_record" ).insertData( {
	  security_user  = userId
	, logged_in_date = Now()
} );
_getCookieService().setVar( name=_getTwoFactorAuthCookiePersistKey(), value=loginRecordId );
```